### DISCUSSION TITLE
AutocompleteArrayInput defaultValue not working

### DISCUSSION THREAD

**Discussion by jogibs** [prisma/prisma#14023] [Q&A] [answered]
I have a custom dashboard page, where I save certain IDs for a resource and save them in local storage, and the code looks something like this:

```javascriot
const MyDialog = () => {
  const [marketIds, setMarketIds] = useLocalStorage<string[]>('marketIds', []);

  return (
    <Dialog>
      <DialogContent>
        <SimpleForm toolbar={false}>
          <ReferenceArrayInput source="markets" reference="Market">
            <AutocompleteArrayInput
              optionText="name"
              onChange={(values) => {
                setMarketIds(values);
              }}
              defaultValue={marketIds}
              fullWidth
            />
          </ReferenceArrayInput>
        </SimpleForm>
      </DialogContent>
    </Dialog>

  )
}
```

Now, the thing is, this code works fine if the marketIds are entries from one of the first 25 items that it pulls once it's mounted, but if it's not, then the markets input is just blank.

My question is: is there a way to preload the defaultValue entries on mount?

One way I've tried it was with passing the `filter={{ ids: marketIds }}` to `<ReferenceArrayInput />`, but this filter is obviously static, so that would affect all subsequent autocomplete queries

I'm using react-admin version 4.1.2

**Comment by austincrim** [comment, marked as answer]
Hey @jogibs 👋,

This repo is about the [Prisma ORM database tool](https://prisma.io), your question does not seem relevant.

Are you looking for a different open-source repo?

**Reply by jogibs** [reply]
Oh, you are right, sorry! I had both projects open and I've accidentally posted the issue in the wrong one!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
